proposals/nnnn-safe-shared-contiguous-storage.md

```diff
@@ -0,0 +1,1232 @@
+    owner: borrowing Owner
+  ) -> dependsOn(owner) Self?
+
+  /// Unsafely create a `Span` over initialized memory.
+  ///
+  /// The memory representing `count` instances starting at
+  /// `pointer` must be owned by the instance `owner`,
+  /// meaning that as long as `owner` is alive the memory will remain valid.
+  ///
+  /// - Parameters:
+  ///   - pointer: a pointer to the first initialized element.
+  ///   - count: the number of initialized elements in the view.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `Span`.
+  public init<Owner>(
+    unsafeStart pointer: UnsafePointer<Element>,
+    count: Int,
+    owner: borrowing Owner
+  ) -> dependsOn(owner) Self
+}
+
+extension Span where Element: BitwiseCopyable {
+
+  /// Unsafely create a `Span` over initialized memory.
+  ///
+  /// The memory in `unsafeBytes` must be owned by the instance `owner`
+  /// meaning that as long as `owner` is alive the memory will remain valid.
+  ///
+  /// `unsafeBytes` must be correctly aligned for accessing
+  /// an element of type `Element`, and must contain a number of bytes
+  /// that is an exact multiple of `Element`'s stride.
+  ///
+  /// - Parameters:
+  ///   - unsafeBytes: a buffer to initialized elements.
+  ///   - type: the type to use when interpreting the bytes in memory.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `Span`.
+  public init<Owner>(
+    unsafeBytes buffer: UnsafeRawBufferPointer,
+    owner: borrowing Owner
+  ) -> dependsOn(owner) Self
+
+  /// Unsafely create a `Span` over a span of initialized memory.
+  ///
+  /// The memory representing `count` instances starting at
+  /// `unsafeRawPointer` must be owned by the instance `owner`,
+  /// meaning that as long as `owner` is alive the memory will remain valid.
+  ///
+  /// `unsafeRawPointer` must be correctly aligned for accessing
+  /// an element of type `Element`.
+  ///
+  /// - Parameters:
+  ///   - unsafeRawPointer: a pointer to the first initialized element.
+  ///   - type: the type to use when interpreting the bytes in memory.
+  ///   - count: the number of initialized elements in the view.
+  ///   - owner: a binding whose lifetime must exceed that of
+  ///            the newly created `Span`.
+  public init<Owner>(
+    unsafeStart pointer: UnsafeRawPointer,
+    byteCount: Int,
+    owner: borrowing Owner
+  ) -> dependsOn(owner) Self
+}
+```
+
+#####  `Collection`-like API:
+
+The following typealiases, properties, functions and subscripts have direct counterparts in the `Collection` protocol hierarchy. Their semantics shall be as described where they counterpart is declared (in `Collection` or `RandomAccessCollection`). The only difference with their counterpart should be a lifetime dependency annotation where applicable, allowing them to return borrowed nonescapable values or borrowed noncopyable values.
+
+```swift
+extension Span {
```

Comment:
```suggestion
extension Span where Element: ~Copyable {
```

Reply:
oh hey, that's every single `Span` extension in the document.

Reply:
fixed the `Span` extensions to all either support non-copyable element types or require `BitwiseCopyable` elements.